### PR TITLE
Add ccpa

### PR DESCRIPTION
**Pull request by jeteve**
## What does this change?
Adds CCPA 'Restrict Data Processing' enforcement to Googletag when the uspapi is on page.

## How to test
Use the provided client.html file provided and make sure you force your ccpaApplies cookie to be true.

### CODE REVIEW

File: ad-gen.coffee [hunk 1]
```diff
@@ -71,10 +71,24 @@ pageLevelCodePart1 = "
 "
 
 pageLevelCodePart2 = "
+  function enforceGTPCCPA(){
+    if(! window.__uspapi ){ return; }
+    window.__uspapi('getUSPData', 2,
+      function( data, success ){
+        if( !success ){ return; }
+        if( data.uspString.match(/^2.Y/gi) ){
```
Comment (mxdvl) [inline ad-gen.coffee]: We've used the following pattern in our CMP, which may be a little clearer? Not sure if this works in CoffeeScript.

```suggestion
        if( data.uspString.charAt(2) === 'Y' ){
```

File: client.html [hunk 2]
```diff
@@ -7,14 +7,52 @@
...
+    <script>
+      (function () { var e = false; var c = window; var t = document; function r() { if (!c.frames["__uspapiLocator"]) { if (t.body) { var a = t.body; var e = t.createElement("iframe"); e.style.cssText = "display:none"; e.name = "__uspapiLocator"; a.appendChild(e) } else { setTimeout(r, 5) } } } r(); function p() { var a = arguments; __uspapi.a = __uspapi.a || []; if (!a.length) { return __uspapi.a } else if (a[0] === "ping") { a[2]({ gdprAppliesGlobally: e, cmpLoaded: false }, true) } else { __uspapi.a.push([].slice.apply(a)) } } function l(t) { var r = typeof t.data === "string"; try { var a = r ? JSON.parse(t.data) : t.data; if (a.__cmpCall) { var n = a.__cmpCall; c.__uspapi(n.command, n.parameter, function (a, e) { var c = { __cmpReturn: { returnValue: a, success: e, callId: n.callId } }; t.source.postMessage(r ? JSON.stringify(c) : c, "*") }) } } catch (a) { } } if (typeof __uspapi !== "function") { c.__uspapi = p; __uspapi.msgHandler = l; c.addEventListener("message", l, false) } })();
+    </script>
+
+    <script>
+      window._sp_ccpa = {
+          config: {
+              mmsDomain: 'https://message.sp-prod.net',
```
Comment (mxdvl) [inline client.html:20]: Would we want to use our own domain?

```suggestion
              mmsDomain: 'https://consent.theguardian.com',
```

File: client.html [hunk 3]
```diff
@@ -7,14 +7,52 @@
...
+              ccpaOrigin: 'https://ccpa-service.sp-prod.net',
+              accountId: 1257,
+              getDnsMsgMms: true,
+              alwaysDisplayDns: false,
+              siteHref: 'http://theguardian.eteve.net',
```
Comment (mxdvl) [inline client.html]: This feels a little personal…

File: client.html [hunk 4]
```diff
@@ -7,14 +7,52 @@
...
+              targetingParams: {
+                  myctype: 'ccpa'
```
Comment (mxdvl) [inline client.html]:
```suggestion
                  framework: 'ccpa'
```

File: ad-gen.coffee (hunk 1 above)
Comment (jeteve) [reply ad-gen.coffee]: 👍

**Review by mxdvl — CHANGES_REQUESTED** [review]

File: client.html (hunk 2 above)
Comment (jeteve) [reply client.html:20]: This file's only for local testing, better keep it as close to standard as possible.

File: client.html (hunk 3 above)
Comment (jeteve) [reply client.html]: I'll try with the general www ccpa setting.

**Review by mxdvl — APPROVED** [review]
👍 LGTM

**Pull request merged**